site/docs/spark-ddl.md

```diff
@@ -182,6 +182,29 @@ ALTER TABLE prod.db.sample
 ADD COLUMN point.z double
 ```
 
+```sql
+-- create a nested array column of struct
+ALTER TABLE prod.db.sample
+ADD COLUMN points array<struct<x: double, y: double>>;
+
+-- add a field to the struct with in array
+ALTER TABLE prod.db.sample
+ADD COLUMN points.element.z double
+```
+
+```sql
+-- create a map column of struct key and struct value
+ALTER TABLE prod.db.sample
+ADD COLUMN points map<struct<x: int>, struct<a: int>>;
+
+-- add a field to the value struct in a map
+ALTER TABLE prod.db.sample
+ADD COLUMN points.value.b int
+```
+
+Note that we cannot perform add columns to the struct 'key' of map column. Only possible for struct 'value' column. 
```

Comment:
In documentation, there is no "we". In most cases you can remove it to be more direct:
> Note: Altering a map column by adding columns is not allowed. Only map values can be updated.

Reply:
Fixed.